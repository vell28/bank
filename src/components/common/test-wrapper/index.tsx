import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { mainDarkTheme } from 'modules/themes';
import { store } from 'modules/store';

interface ITestWrapper {
  children: React.ReactNode;
}

export const TestWrapper: React.FC<ITestWrapper> = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={mainDarkTheme}>
      <>{children}</>
    </ThemeProvider>
  </Provider>
);
