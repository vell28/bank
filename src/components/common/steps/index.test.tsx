import React from 'react';

import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { TestWrapper } from '../test-wrapper';

import { Step, Steps } from '.';

describe('<Steps /> tests', () => {
  test('Should success render <Steps />, should render step of id', () => {
    const component: any = mount(
      <TestWrapper>
        <Steps stepId={0}>
          <Step id={0}>
            <span className="step-0">1</span>
          </Step>
          <Step id={1}>
            <span className="another">1</span>
          </Step>
          <Step id={2}>2</Step>
        </Steps>
      </TestWrapper>,
    );
    const activeStep = component.find('.step-0');
    const anotherStep = component.find('.anotherStep');

    expect(activeStep.length).toEqual(1);
    expect(anotherStep.length).toEqual(0);
    expect(toJson(component)).toMatchSnapshot();
  });
});
