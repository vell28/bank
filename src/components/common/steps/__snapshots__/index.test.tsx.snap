// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Steps /> tests Should success render <Steps />, should render step of id 1`] = `
<TestWrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "breakpoints": Array [
            "576px",
            "768px",
            "992px",
            "1200px",
          ],
          "colors": Object {
            "alto": "#dedede",
            "asparagus": "#77a45c",
            "black": "#000000",
            "blue": "#008cff",
            "blueLight": "#70cde4",
            "borderLight": "#cad3db",
            "dark": "#030303",
            "darkBlue": "#011f3d",
            "darkBlue2": "#132F4B",
            "elephant": "#12304b",
            "feijoa": "#93d77e",
            "gray": "#a5adb8",
            "gray10": "#cad3db",
            "gray11": "#959fac",
            "gray2": "#a7a9ac",
            "gray3": "#a1aab6",
            "gray4": "#e8e8e8",
            "gray5": "#cbcbcb",
            "gray6": "#eaeaea",
            "gray7": "#acacac",
            "gray9": "#9faab7",
            "grayHover": "#00000008",
            "grayLight": "#cdd0d5",
            "green": "#7ed321",
            "green1": "#5ba00f",
            "green2": "#4dd964",
            "green3": "#77a45c",
            "green4": "#b1de6b",
            "greenLight": "#d9f9b6",
            "light": "#d5dae0",
            "malibu": "#53b8ff",
            "navy": "#01172d",
            "orange": "#e4be79",
            "persimmon": "#ff5353",
            "purple": "#a479cb",
            "red": "#e10606",
            "red2": "#ff2e7b",
            "red3": "#ea4848",
            "rose": "#d08484",
            "slate": "#596778",
            "submarine": "#b6bcc4",
            "turquoise": "#79cbcb",
            "white": "#ffffff",
            "white2": "#dddddd",
          },
          "dimension": Object {},
          "fonts": Object {
            "faFontsPath": "/css/webfonts",
            "fontFamily": "AvenirNext, Arial",
          },
          "name": "main-dark-theme",
          "size": "default",
          "typography": Object {
            "bodyBackground": "#011f3d",
            "bodyFontColor": "#ffffff",
            "bodyFontFamily": "AvenirNext, Arial",
            "globalFontSize": "1.6rem",
            "globalLineHeight": 1.5,
            "globalWeightNormal": 400,
            "headersFontColor": "#ffffff",
            "headersFontFamily": "AvenirNext, Arial",
            "headersFontWeight": 700,
          },
        }
      }
    >
      <Memo()
        stepId={0}
      >
        <Memo()
          id={0}
        >
          <span
            className="step-0"
          >
            1
          </span>
        </Memo()>
      </Memo()>
    </ThemeProvider>
  </Provider>
</TestWrapper>
`;
